S_Z for wall tool to change_extrusion_depth

falken10vdl opened the pull request:
This is a proposal to allow to extend walls in the z direction similar to how it is done with S_E

falken10vdl committed:
S_Z for wall tool to change_extrusion_depth

falken10vdl committed:
changed from S_Z to C_E

theoryshaw commented:
This is awesome. 
Would it be possible for this to work on all LAYER2 objects for all classes?
For example, i could see using this tool for IfcCoverings as well, among other classes.

falken10vdl committed:
remove restriction for ifcWall. just allow for LAYER2

falken10vdl commented:
@theoryshaw I have removed the ifcWall restriction and do bim.change_extrusion_depth for all LAYER2 objects.
Sorry I do not know exactly how these LAYER2 obejcts work :) I need to go over your tutorials to have a clearer picture.
By looking to the code I see the ChangeExtrusionDepth seems to be able to be applied to LAYER2.
I have added an ifccovering with IfcMaterialLayerUsage so it has an AXIS2 (and therefore tool.Model.get_usage_type(element) returns "LAYER2") and it is properly handled. Is this what you are looking for?
<img width="1062" height="736" alt="image" src="https://github.com/user-attachments/assets/c17558a0-53ef-4d0f-8065-83363464aa3e" />
and after C_E...
<img width="1057" height="758" alt="image" src="https://github.com/user-attachments/assets/11f1321e-8765-4d2b-9b8c-3262ba93e9d8" />

Cheers!

theoryshaw commented:
Works great! thank you.

brunoperdigao commented:
@falken10vdl Nice addition! I guess is just missing that you run black formatter and we're good to go.

falken10vdl committed:
black formatting

falken10vdl commented:
Cool! thanks @brunoperdigao!  I just run black formatting.
Cheers!
